Tests/MuxUploadSDKTests/Upload Tests/UploadPersistenceTests.swift

```diff
@@ -62,14 +65,24 @@ final class UploadPersistenceTests: XCTestCase {
             uploadInfo: renameDummyUploadInfo(basename: "e2")
         )
         let persistence = UploadPersistence(innerFile: makeSimulatedUploadsFile(), atURL: makeDummyFileURL(basename: "fake-cache-file"))
-        // Shouldn't throw (but not specifically part of the test)
-        try! persistence.write(entry: e1, forFileAt: makeDummyFileURL(basename: "e1"))
-        try! persistence.write(entry: e2, forFileAt: makeDummyFileURL(basename: "e2"))
+        XCTAssertNoThrow(
+            try persistence.write(entry: e1, for: e1.uploadInfo.id)
+        )
+
+        XCTAssertNoThrow(
+            try persistence.write(entry: e2, for: e2.uploadInfo.id)
+        )
         
         // Read them in a different order to ensure idempotence
-        let readItem1 = try persistence.readEntry(forFileAt: makeDummyFileURL(basename: "e2"))
-        let readItem2 = try persistence.readEntry(forFileAt: makeDummyFileURL(basename: "e1"))
-        let allItems = try persistence.readAll()
+        let readItem1 = try XCTUnwrap(
+            persistence.readEntry(uploadID: e2.uploadInfo.id)
+        )
+        let readItem2 = try XCTUnwrap(
```

Comment:
Converts what is otherwise a crash into a failing test. More here: https://useyourloaf.com/blog/xctunwrap-optionals-in-your-tests/